### DISCUSSION TITLE
Meaning of Interaction Terms of Pairwise T tests for Mixed Designs

### DISCUSSION THREAD

**Discussion by Ruairi-osul** [raphaelvallat/pingouin#277] [Q&A] [answered]
Hello,

I have a quite basic question. When running the pairwise pg.pairwise_ttests for mixed designs, why are there interaction terms for each level of the within subject effect? 

For example, the below shows the output of the function. My intuition is that the last three red contrasts are the inverse of the previous three green contrasts. I am misunderstanding what the contrasts represent?

<img width="457" alt="image" src="https://user-images.githubusercontent.com/29435612/174394224-9cf2d610-4111-4821-bd94-2a65298fca23.png">

**Comment by raphaelvallat** [comment, marked as answer]
Hi @Ruairi-osul,

By default, Pingouin does not report all the possible pairs of tests like other softwares such as JASP (e.g. Pre FF vs post SIR, etc). Instead, it takes each level of the within variable separately, and then compare all the levels of the between variable (Pre - FF vs SIR, Pre - FF vs SR, Post - FF vs SR, etc). You can also do the opposite by passing `within_first=False`, in which case Pingouin will take each level of the between variable separately and then compare the within level (FF - pre vs post, SIR - pre vs post, SR - pre vs post).

Thanks,
Raphael

**Comment by Ruairi-osul** [comment]
Brilliant. Thanks Raphael. That makes perfect sense. Appreciate your answer and work :)
